### PR TITLE
Use tidb_query_vec_executors instead of normal executors

### PR DESCRIPTION
**Pull request by Renkai**
### What problem does this PR solve?

Issue Number: https://github.com/tikv/tikv/issues/7039

Problem Summary: migrate some normal executors to vector  executors.

### Related changes

- Nope
### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Nope

### Release note

### CODE REVIEW

**Comment by Renkai** [comment]
@zhongzc @iosmanthus @breeswish PTAL

File: src/coprocessor/statistics/analyze.rs [hunk 1]
```diff
@@ -73,27 +77,48 @@ impl<S: Snapshot> AnalyzeContext<S> {
...
-            let row = row.take_origin()?;
-            let (bytes, end_offsets) = row.data.get_column_values_and_end_offsets();
-            hist.append(bytes);
-            if let Some(c) = cms.as_mut() {
-                for end_offset in end_offsets {
-                    c.insert(&bytes[..end_offset])
+
+        let mut is_drained = false;
+        let mut batch_size = BATCH_INITIAL_SIZE;
+        while !is_drained {
```
Comment (breeswish) [inline src/coprocessor/statistics/analyze.rs:94]: I guess there is no need to follow a grow strategy for analyze

File: src/coprocessor/statistics/analyze.rs [hunk 2]
```diff
@@ -73,27 +77,48 @@ impl<S: Snapshot> AnalyzeContext<S> {
...
+
+            let batch = scanner.next_batch(batch_size);
+            is_drained = batch.is_drained?;
+
+            for logical_row in batch.logical_rows {
+                let mut bytes = vec![];
+                for col in batch.physical_columns.as_slice() {
+                    let buffer_vec = col.raw();
+                    let data = buffer_vec.index(logical_row);
+                    bytes.extend_from_slice(data);
```
Comment (breeswish) [inline src/coprocessor/statistics/analyze.rs:105]: Looks like the behaviour is scanning data from row, convert to column, and here convert it back to row?

File: src/coprocessor/statistics/analyze.rs (hunk 1 above)
Comment (Renkai) [reply src/coprocessor/statistics/analyze.rs:94]: > I guess there is no need to follow a grow strategy for analyze

So is there a suitable constant for `batch_size`?

File: src/coprocessor/statistics/analyze.rs (hunk 2 above)
Comment (Renkai) [reply src/coprocessor/statistics/analyze.rs:105]: > Looks like the behaviour is scanning data from row, convert to column, and here convert it back to row?

I think a more suitable description is scanner scanned a matrix, then here get every row from matrix.

**Comment by Renkai** [comment]
@sre-bot /run-all-tests

**Comment by Renkai** [comment]
@sre-bot /release make dist_release

**Comment by Renkai** [comment]
@sre-bot /bench +sysbench +tpcc +tpch

**Comment by Renkai** [comment]
@sre-bot /run-all-tests

**Comment by Renkai** [comment]
@ti-srebot bot /run-all-tests

**Comment by Renkai** [comment]
@ti-srebot /release make dist_release

**Comment by Renkai** [comment]
@ti-srebot /bench +sysbench +tpcc +tpch

**Comment by Renkai** [comment]
@ti-srebot /bench +sysbench

**Comment by Renkai** [comment]
@ti-srebot  /release make dist_release

**Comment by Renkai** [comment]
@ti-srebot /bench +sysbench +tpcc +tpch

**Comment by mahjonp** [comment]
Seems that tpch q22 always hangs when benching on the master branch..

**Comment by Renkai** [comment]
No need to continue because of https://github.com/tikv/tikv/pull/8215

**Pull request closed without merging**